src/Files.App/Data/Models/ContextFlyoutItemModel.cs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2023 Files Community
+// Licensed under the MIT License. See the LICENSE.
+
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media.Imaging;
+using System.Windows.Input;
+using Windows.System;
+
+namespace Files.App.Data.Models
+{
+	public sealed class ContextFlyoutItemModel
+	{
+		public ContextFlyoutItemType ItemType { get; set; }
+
+		public ICommand? Command { get; set; }
+
+		public object? CommandParameter { get; set; }
+
+		public string? Text { get; set; }
+
+		public BitmapImage? BitmapIcon { get; set; }
+
+		public string? Glyph { get; set; }
+
+		public string? GlyphFontFamilyName { get; set; }
+
+		public KeyboardAccelerator? KeyboardAccelerator { get; set; }
+
+		public string? KeyboardAcceleratorTextOverride { get; set; }
+
+		public object? Tag { get; set; }
+
+		public Func<Task>? LoadSubMenuAction { get; set; }
+
+		public string? ID { get; set; }
+
+		public OpacityIconModel? OpacityIcon { get; set; }
+
+		public List<ContextFlyoutItemModel>? Items { get; set; }
+
+		public bool IsAvailable { get; set; } = true;
+		public bool IsVisible { get; set; } = true;
+		public bool IsChecked { get; set; }
+		public bool IsEnabled { get; set; } = true;
+		public bool IsPrimary { get; set; }
+
+		public bool ShowOnShift { get; set; }
+		public bool ShowInRecycleBin { get; set; }
+		public bool ShowInSearchPage { get; set; }
+		public bool ShowInFtpPage { get; set; }
+		public bool ShowInZipPage { get; set; }
+
+		public bool SingleItemOnly { get; set; }
+
+		public bool CollapseLabel { get; set; }
+
+		public bool ShowLoadingIndicator { get; set; }
+
+		/// <summary>
+		/// Initializes an instance of <see cref="ContextFlyoutItemModel"/>.
+		/// </summary>
+		public ContextFlyoutItemModel(ContextFlyoutItemType type = ContextFlyoutItemType.Button)
+		{
+			ItemType = type;
+		}
+
+		/// <summary>
+		/// Initializes an instance of <see cref="ContextFlyoutItemModel"/> with <see cref="IRichCommand"/>.
+		/// </summary>
+		/// <param name="richCommand"><see cref="IRichCommand"/> instance.</param>
+		/// <param name="richCommandParameter"><see cref="IRichCommand"/> parameter.</param>
+		public ContextFlyoutItemModel(ContextFlyoutItemType type, IRichCommand richCommand, object? richCommandParameter = null) : this(type)
+		{
+			Command = richCommand;
+
+			//richCommand.Parameter = richCommandParameter;
+
+			bool isExecutable = richCommand.IsExecutable;
+			if (!isExecutable)
+				return;
+
+			Text = richCommand.Label;
+			IsEnabled = isExecutable;
+			IsChecked = richCommand.IsOn;
+			ItemType = type;
+			IsAvailable = true;
```

Comment:
What does this property do?

Reply:
IsAvailable is the value that indicates whether item should be displayed. If true, it will be converted to an UIElement; otherwise, even never included the return value of the generation method. IsVisible is a temporary solution to hide, once it's hidden and is ready to be displayed again, it will be set true.

Reply:
We need some comments in the code but that's out of scope for now, in the meantime, I find "ShowItem" to be a little more intuitive.

Reply:
I'm probably not gonna touch around here again so adding comments would be a better idea.

Reply:
Done